### PR TITLE
Use types from protocol in VsX

### LINKED ISSUE #742: Refactor vscode to use common types and errors from taqueria protocol
### Describe the feature you would like

In 620-operations, there was a large refactoring effort to reduce code duplication by having many of our custom defined concrete types moved to the taqueria-protocol package, which is shared between the taqueria-cli, taqueria-sdk, and too, the vscode extension. As such, we were able to remove a lot of duplication from the SDK and I suspect we can do the same with the vscode extension.

The 620-operations branch should be merged in some time tomorrow. I hold hold on working on this issue till that occurs. 

### Is this feature solving a problem?

Consistent parsing, validation, and error messaging.

### Please share any additional details you have?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by AlirezaHaghshenas**
closes #742 

We can argue that the previous style of defining the errors is more suitable for the code in VsX
@mweichert Can we have a conversation about this?

### CODE REVIEW

File: taqueria-vscode-extension/src/lib/helpers.ts [hunk 1]
```diff
@@ -156,19 +158,19 @@ export const inject = (deps: InjectedDependencies) => {
 	const taskNameToCmdId = (taskName: string) => 'taqueria.' + taskName.replace(/\s+/g, '_')
 
 
-	const exposeTasksFromProject = (context: api.ExtensionContext, output: api.OutputChannel, folders: readonly api.WorkspaceFolder[], i18n: I18N) =>
+	const exposeTasksFromProject = (context: api.ExtensionContext, output: api.OutputChannel, folders: readonly api.WorkspaceFolder[], i18n: i18n) =>
 		(projectDir: Util.PathToDir) => {
 			getTaqBinPath(i18n)
 			.then(pathToTaq => Util.proxyToTaq(pathToTaq, i18n, projectDir) ('list-known-tasks'))
-			.then(data => Util.decodeJson<Util.State>(data))
+			.then(data => Util.decodeJson<PersistentState>(data))
```
Comment (AlirezaHaghshenas) [inline taqueria-vscode-extension/src/lib/helpers.ts:165]: This should be EphemeralState

File: taqueria-vscode-extension/src/lib/pure.ts [hunk 2]
```diff
@@ -94,24 +22,15 @@ export type PathToFile = string & {__kind__: 'PathToFile'}
...
-
 export const taqifiedDirType: unique symbol = Symbol("taqifiedDirType")
 
 export class TaqifiedDir {
     [taqifiedDirType]: void
     readonly dir: PathToDir
-    readonly config: Config
-    readonly state: State
+    readonly config: Config.t
+    readonly state: PersistentState
```
Comment (AlirezaHaghshenas) [inline taqueria-vscode-extension/src/lib/pure.ts:31]: Same here

File: taqueria-vscode-extension/src/lib/pure.ts [hunk 3]
```diff
@@ -259,7 +178,7 @@ export const proxyToTaq = (pathToTaq: PathToTaq, i18n: I18N, projectDir?: PathTo
     .catch(previous => {
         if ("code" in previous) {
             if (previous.code === 'E_EXEC') {
-                const {cmd, msg} = previous as E_EXEC
+                const {cmd, msg} = previous
```
Comment (AlirezaHaghshenas) [inline taqueria-vscode-extension/src/lib/pure.ts:181]: This code path might not be needed anymore

**Review by hu3man — APPROVED** [review]
LGTM

**Review by jchenche — APPROVED** [review]

**Pull request merged**